lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1484,9 +1484,9 @@ def deploy_vm(self):
         else:
             vm_obj = None
 
-        # need a resource pool if cloning from template
-        if self.params['resource_pool'] or self.params['template']:
-            resource_pool = self.get_resource_pool()
+        resource_pool = None
```

Comment:
No need to set resource_pool to None. `get_resource_pool` return None if no resource pool found.

Reply:
Removed from the code.